contrib/tools/dummysigner/src/sign.rs

```diff
@@ -90,41 +94,152 @@ impl Signer {
         Ok(())
     }
 
-    pub fn sign_revocation_txs(
+    pub fn derive_revocation_txs(
         &self,
         outpoint: OutPoint,
         amount: Amount,
         derivation_index: ChildNumber,
     ) -> Result<RevocationTransactions, Error> {
-        let descriptors = self.descriptors.unwrap_or(Err(Error(
-            "Wallet does not have the descriptors".to_string(),
-        )))?;
-        let emer_address = self.emergency_address.unwrap_or(Err(Error(
-            "Wallet does not have emergency_address".to_string(),
-        )))?;
-        let (_, cancel_tx, emergency_tx, unvault_emergency_tx) = transaction_chain(
+        let descriptors = self
+            .descriptors
+            .as_ref()
+            .ok_or(Error("Wallet does not have the descriptors".to_string()))?;
+        let emer_address = self
+            .emergency_address
+            .as_ref()
+            .ok_or(Error("Wallet does not have emergency_address".to_string()))?;
+        let (_, cancel_tx, emergency_tx, emergency_unvault_tx) = transaction_chain(
             outpoint,
             amount,
             &descriptors.deposit_descriptor,
             &descriptors.unvault_descriptor,
             &descriptors.cpfp_descriptor,
             derivation_index,
-            emer_address,
+            emer_address.clone(),
             0,
             &self.curve,
         )
         .map_err(|e| Error(e.to_string()))?;
 
         Ok(RevocationTransactions {
-            cancel_tx,
-            emergency_tx,
-            unvault_emergency_tx,
+            cancel_tx: cancel_tx.into_psbt(),
+            emergency_tx: emergency_tx.into_psbt(),
+            emergency_unvault_tx: emergency_unvault_tx.into_psbt(),
         })
     }
+
+    pub fn derive_unvault_tx(
+        &self,
+        outpoint: OutPoint,
+        amount: Amount,
+        derivation_index: ChildNumber,
+    ) -> Result<PartiallySignedTransaction, Error> {
+        let descriptors = self
+            .descriptors
+            .as_ref()
+            .ok_or(Error("Wallet does not have the descriptors".to_string()))?;
+
+        let deposit_descriptor = descriptors
+            .deposit_descriptor
+            .derive(derivation_index, &self.curve);
+        let deposit_txin =
+            DepositTxIn::new(outpoint, DepositTxOut::new(amount, &deposit_descriptor));
+        let unvault_descriptor = descriptors
+            .unvault_descriptor
+            .derive(derivation_index, &self.curve);
+        let cpfp_descriptor = descriptors
+            .cpfp_descriptor
+            .derive(derivation_index, &self.curve);
+
+        let unvault_tx =
+            UnvaultTransaction::new(deposit_txin, &unvault_descriptor, &cpfp_descriptor, 0)
```

Comment:
I'm not sure it's fine to sign with locktime 0 😅

Reply:
https://github.com/revault/revaultd/blob/14463bc01840d99c82a4b474b2f649cbd90182de/src/daemon/revaultd.rs#L261
Should I add it to the config ?

Reply:
Ah, I didn't know! TIL :) 

It's fine as it this then :)